lib/apple_sign_in/user_identity.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: false
+
+module AppleSignIn
+  class UserIdentity
+    APPLE_KEY_URL = 'https://appleid.apple.com/auth/keys'.freeze
+
+    attr_reader :user_identify, :jwt
+
+    def initialize(user_identify, jwt)
+      @user_identify = user_identify
+      @jwt = jwt
+    end
+
+    def valid?
+      token_data = apple_token_from_jwk
+
+      return true if token_data.key?('sub') && user_identify == token_data['sub']
```

Comment:
The 'sub' short name can be a bit cryptic for those not familiarised with jwt. Maybe a refactor extracting the logical expression to

```ruby
def match_user_identity?
  token_data.key?('sub') && user_identify == token_data['sub']
end
```

could be more intention revealing